pkg/controller/deployment/deployment_controller.go

```diff
@@ -22,79 +22,314 @@ import (
 	"time"
 
 	"github.com/golang/glog"
+
 	"k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/apis/extensions"
+	"k8s.io/kubernetes/pkg/client/cache"
 	"k8s.io/kubernetes/pkg/client/record"
 	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/controller"
+	"k8s.io/kubernetes/pkg/controller/framework"
 	"k8s.io/kubernetes/pkg/fields"
 	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/runtime"
 	"k8s.io/kubernetes/pkg/util"
 	deploymentutil "k8s.io/kubernetes/pkg/util/deployment"
+	"k8s.io/kubernetes/pkg/util/workqueue"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+const (
+	// FullDeploymentResyncPeriod means we'll attempt to recompute the required replicas
+	// of all deployments that have fulfilled their expectations at least this often.
+	// This recomputation happens based on contents in the local caches.
+	FullDeploymentResyncPeriod = 30 * time.Second
```

Comment:
https://github.com/kubernetes/kubernetes/pull/15153 removed this from all existing controllers at that time and replaced it by 12 hour unified period for all controllers.
cc @wojtek-t